src/python_bindings/bindings.cpp

```diff
@@ -368,87 +367,71 @@ class PyTieredIndex : public PyVecSimIndex {
 private:
     VecSimIndexAbstract<float> *getFlatBuffer() {
         return reinterpret_cast<VecSimTieredIndex<float, float> *>(this->index.get())
-            ->getFlatbufferIndex();
-    }
-
-protected:
-    JobQueue jobQueue;       // External queue that holds the jobs.
-    IndexExtCtx jobQueueCtx; // External context to be sent to the submit callback.
-    SubmitCB submitCb;       // A callback that submits an array of jobs into a given jobQueue.
-    size_t flatBufferLimit; // Maximum size allowed for the flat buffer. If flat buffer is full, use
-                            // in-place insertion.
-    bool run_thread;
-    std::bitset<MAX_POOL_SIZE> executions_status;
-
-    TieredIndexParams TieredIndexParams_Init() {
-        TieredIndexParams ret = {
-            .jobQueue = &this->jobQueue,
-            .jobQueueCtx = &this->jobQueueCtx,
-            .submitCb = this->submitCb,
-            .flatBufferLimit = this->flatBufferLimit,
-        };
-
-        return ret;
+            ->getFlatBufferIndex();
     }
 
 public:
-    explicit PyTieredIndex(size_t BufferLimit = 3000000)
-        : submitCb(submit_callback), flatBufferLimit(BufferLimit), run_thread(true) {
-
-        for (size_t i = 0; i < THREAD_POOL_SIZE; i++) {
-            ThreadParams params(run_thread, executions_status, i, jobQueue);
-            thread_pool.emplace_back(thread_main_loop, params);
+    explicit PyTieredIndex() {
+        if (tieredIndexMock::ctx != nullptr || !tieredIndexMock::jobQ.empty() ||
+            tieredIndexMock::thread_pool.size() > 0) {
+            throw std::runtime_error("Previous tests did not ended successfully - aborting test");
```

Comment:
Yes, if the test doesn't pass the distractor will not be invoked (I checked). In that case, I don't think that we should proceed because we cannot always clean up properly and continue...